BraveRewardsUI/SKUs/SKUPurchaseViewController.swift

```diff
@@ -0,0 +1,139 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import BraveRewards
+import BraveShared
+
+public class SKUPurchaseViewController: UIViewController, UIViewControllerTransitioningDelegate {
+  
+  private let rewards: BraveRewards
+  private let publisher: PublisherInfo
+  private let request: PaymentRequest
+  private let ledgerObserver: LedgerObserver
+  private var openBraveTermsOfSale: () -> Void
+  private let responseHandler: (_ response: PaymentRequestResponse) -> Void
+  
+  public init(
+    rewards: BraveRewards,
+    publisher: PublisherInfo,
+    request: PaymentRequest,
+    responseHandler: @escaping (_ response: PaymentRequestResponse) -> Void,
+    openBraveTermsOfSale: @escaping () -> Void
+  ) {
+    self.rewards = rewards
+    self.publisher = publisher
+    self.request = request
+    self.responseHandler = responseHandler
+    self.openBraveTermsOfSale = openBraveTermsOfSale
+    self.ledgerObserver = LedgerObserver(ledger: rewards.ledger)
+    
+    super.init(nibName: nil, bundle: nil)
+    
+    modalPresentationStyle = .overCurrentContext
+    if #available(iOS 13.0, *) {
+      isModalInPresentation = true
+    }
+    transitioningDelegate = self
+    
+    self.rewards.ledger.add(self.ledgerObserver)
+    setupLedgerObserver()
+    
+    rewards.ledger.fetchBalance(nil)
+  }
+  
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+  
+  private var purchaseView: SKUPurchaseView {
+    return view as! SKUPurchaseView // swiftlint:disable:this force_cast
+  }
+  
+  public override func loadView() {
+    view = SKUPurchaseView()
+  }
+  
+  public override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    title = Strings.SKUPurchaseTitle
+    
+    purchaseView.detailView.itemDetailValueLabel.text = request.details.displayItems.map { $0.label }.joined(separator: ", ")
```

Comment:
This label originates from developers, I tried a couple of strings but wasn't able to execute scripts using UILabel text. Not sure if we need to sanitize the text before showing it on the payment window.

Reply:
As far as I know, we don't need to do any sanitation. Not sure if we should clean it up though  (remove whitespace, etc.)